### ISSUE TITLE
When the data's changed for the line Chart, the gradient background on hover stay in opaque gray

### ISSUE THREAD

**Issue by beubzz** [swimlane/ngx-charts#1158]
- [x ] bug report - search github for a similar issue or PR before submitting
- [ ] feature request
- [ ] support request - use StackOverflow (add the `ngx-charts` tag) or the gitter chat for support questions

**Current behavior**
When the data's changed, the gradient background on hover stay in opaque gray (very dirty).
The linearGradient doesn't persist with the good color and gradient when the data's changed ...

**Expected behavior**
When the data's changed, i would like a gradient background with the good color like i declare on colorScheme.

**Reproduction of the problem**
When the data's changed, the linearGradient Id will be undefined but why ?
![linear-charts](https://user-images.githubusercontent.com/10169123/57233687-6b6b4500-701f-11e9-8557-b9a24359c6da.png)

On init it's be perfect (on the over, i got a good gradient with good color) : 
![good-hover](https://user-images.githubusercontent.com/10169123/57233981-06641f00-7020-11e9-9b1c-72a9e928fd78.png)

But when i change the data's, i have a bad gradient and color on the hover : 
![bad-hover](https://user-images.githubusercontent.com/10169123/57234006-1b40b280-7020-11e9-9ef5-76c84a59ff91.png)

/!\ IT'S FOR A LINE-CHART /!\

**Please tell us about your environment:**
Windows10 - VisualCode (on local)

* **ngx-charts version:**
@swimlane/ngx-charts: "10.1.0"

* **Browser:**
Chrome

* **Angular version:**
Angular 7.1.3

* **Language:**
TypeScript : 3.1.6

the HTML declaration :  
<ngx-charts-line-chart
        [view]="view"
        [showGridLines]="true"
        [roundDomains]="true"
        [scheme]="colorScheme"
        [results]="selectedData"
        [gradient]="true"
        [xAxis]="true"
        [yAxis]="true"
        [autoScale]="true"
        [timeline]="false"
        [legend]="true"
        [legendPosition]="'right'"
        [showXAxisLabel]="false"
        [showYAxisLabel]="false">

The ts declaration : 
public view: any[] = [1000, 300];
public colorScheme = {
    domain: ['#5AA454', '#A10A28']
};

On my ngOnInit : 
// init the chart data
this.selectedData = new Array(this.data[0]);

when my input change, I fire this function : 
public onChange(event) {
    this.selectedData = new Array(this.data[this.data.findIndex(elem => elem.id === event.id)]);
}

My data's structure be like : 
`
let data = [
      {
        id: '15',
        name: 'Ligne 15',
        series: [
          {
            name: '06h',
            value: 5
          },
          {
            name: '07h',
            value: 9
          },
          {
            name: '08h',
            value: 15
          },
          {
            name: '09h',
            value: 22
          },
          {
            name: '10h',
            value: 25
          },
          {
            name: '11h',
            value: 27
          },
          {
            name: '12h',
            value: 22
          },
          {
            name: '13h',
            value: 21
          },
          {
            name: '14h',
            value: 25
          },
          {
            name: '15h',
            value: 27
          },
          {
            name: '16h',
            value: 28
          },
          {
            name: '17h',
            value: 28
          },
          {
            name: '18h',
            value: 35
          },
          {
            name: '19h',
            value: 22
          },
          {
            name: '20h',
            value: 14
          },
          {
            name: '21h',
            value: 11
          },
          {
            name: '22h',
            value: 3
          },
        ]
      },
      {
        id: '11',
        name: 'Ligne 11',
        series: [
          {
            name: '06h',
            value: 2
          },
          {
            name: '07h',
            value: 5
          },
          {
            name: '08h',
            value: 8
          },
          {
            name: '09h',
            value: 12
          },
          {
            name: '10h',
            value: 15
          },
          {
            name: '11h',
            value: 15
          },
          {
            name: '12h',
            value: 12
          },
          {
            name: '13h',
            value: 12
          },
          {
            name: '14h',
            value: 17
          },
          {
            name: '15h',
            value: 17
          },
          {
            name: '16h',
            value: 19
          },
          {
            name: '17h',
            value: 22
          },
          {
            name: '18h',
            value: 22
          },
          {
            name: '19h',
            value: 15
          },
          {
            name: '20h',
            value: 11
          },
          {
            name: '21h',
            value: 3
          },
          {
            name: '22h',
            value: 0
          },
        ]
      }
    ];
`

Everythings works very well without the gradient color on the hover (on a point or on the legend it's the same opaque dirty gray ...)

**Comment by marjan-georgiev** [comment]
Could you create a stackblitz where this can be reproduced?

**Comment by xjose97x** [comment]
I can confirm this issue persists.

**Comment by Seguix** [comment]
Can someone solve this? The same thing happens to me.

**Comment by ron2702** [comment]
Hi everyone, I have the same issue.

**Comment by rickymohk** [comment]
Same issue in 2021

**Comment by bobbyg603** [comment]
@marjan-georgiev I have created a [stackblitz](https://stackblitz.com/edit/ngx-charts-line-chart-bug?file=src/app/app.component.ts) that demonstrates this issue. We tried to fix this locally but was unable to get npm link @swimlane/ngx-charts to work in my Angular 12 project.  Any guidance you can give us would be greatly appreciated!

**Comment by bobbyg603** [comment]
@Seguix @xjose97x @ron2702 @rickymohk there's a workaround. it's not great, but it will fix the issue with the gradient,

```ts
@Input()
set data(value) {
    this._data = [];
    setTimeout(() => this._data = value, 150);
}
```

**Comment by bobbyg603** [comment]
@Seguix @xjose97x @ron2702 @rickymohk we've fixed this bug and have published `@bugsplat/ngx-charts` to [npm](https://www.npmjs.com/package/@bugsplat/ngx-charts). additionally, we opened #1631 in hopes of getting this fix merged with the official repo.

**Comment by bobbyg603** [comment]
Upon further inspection, the fix we proposed actually addresses a different problem (and breaks the gradient functionality). This bug actually occurs when the [gradient] input to ngx-charts-line-chart is left as its default value of false. This means that the line fill should be a solid color, but instead it is a solid gray. We found that if you want to use the gradient for ngx-charts-line-chart just set [gradient]="true" and everything works correctly
